packages/autocomplete-plugin-algolia-insights/src/createSearchInsightsApi.ts

```diff
@@ -83,13 +84,17 @@ export function createSearchInsightsApi(searchInsights: InsightsClient) {
      */
     clickedObjectIDsAfterSearch(
       ...params: Array<
-        InsightsParamsWithItems<ClickedObjectIDsAfterSearchParams>
+        WithArbitraryParams<
+          InsightsParamsWithItems<ClickedObjectIDsAfterSearchParams>
+        >
       >
     ) {
       if (params.length > 0) {
         sendToInsights(
           'clickedObjectIDsAfterSearch',
-          mapToInsightsParamsApi(params),
+          mapToInsightsParamsApi<
```

Comment:
The way `mapToInsightsParamsApi` is typed is what's causing the typing issues I was encountering. When I inline the implementation of `mapToInsightsParamsApi` directly, the issue disappears.

I'm not sure what's going on exactly in how TypeScript interprets it all, but when unioning a strict record with an arbitrary record, the generic type `TInsightsParamsType` loses all the mandatory properties and the return type only contains `objectIDs`.

I'm doubtful we can fix this at the `mapToInsightsParamsApi` level, so the pragmatic option seems to cast.

Lmk if you have a working option that works better.